Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLWriter.java

```diff
@@ -0,0 +1,337 @@
+                transformer.setOutputProperty(TRANSFORMER_INDENT_VALUE, Integer.toString(indent));
+            }
+            try (OutputStream out = new FileOutputStream(file)) {
+                transformer.transform(new DOMSource(toDocument(node, namespaces)), new StreamResult(out));
+                out.flush();
+            }
+        } catch (TransformerException e) {
+            String message = e.getMessage();
+            throw new RuntimeException(message != null ? message : e.getClass().getSimpleName(), e);
+        }
+    }
+
+    /**
+     * Outputs this node as an XML document to a String list.
+     *
+     * @param node
+     *            node to convert
+     *
+     * @param indent
+     *            number of spaces of each indent level. A negative number
+     *            disables wrapping
+     * @param namespaces
+     *            an external defined mapping of namespaces to abbreviations
+     * @param charset
+     *            The charset name to be used in the XML declaration. The
+     *            charset is also actively used in the transformation process,
+     *            thus unmappable characters will get lost. Defaults to UTF-8.
+     * @return a String list holding this node as XML
+     */
+    public static List<String> toLines(Node node, int indent, Map<String, String> namespaces,
+            Optional<String> charset) {
+        return Arrays.asList(toString(node, indent, namespaces, charset).split("\r?\n"));
+    }
+
+    /**
+     * Outputs this node as an XML document to a String.
+     *
+     * @param node
+     *            node to convert
+     *
+     * @param indent
+     *            number of spaces of each indent level. A negative number
+     *            disables wrapping
+     * @param namespaces
+     *            an external defined mapping of namespaces to abbreviations
+     * @param optionalCharset
+     *            The charset name to be used in the XML declaration. The
+     *            charset is also actively used in the transformation process,
+     *            thus unmappable characters will get lost. Defaults to UTF-8.
+     * @return a String list holding this node as XML
+     */
+    public static String toString(Node node, int indent, Map<String, String> namespaces,
+            Optional<String> optionalCharset) {
+        String charset = optionalCharset.orElse(StandardCharsets.UTF_8.toString());
+
+        try {
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            if (indent > -1) {
+                transformer.setOutputProperty(OutputKeys.INDENT, TRANSFORMER_INDENT_TRUE);
+                transformer.setOutputProperty(TRANSFORMER_INDENT_VALUE, Integer.toString(indent));
+            }
+            transformer.setOutputProperty(OutputKeys.ENCODING, charset);
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream();
```

Comment:
You are opening a stream without closing it properly.

Reply:
[`ByteArrayOutputStream`](https://docs.oracle.com/javase/8/docs/api/java/io/ByteArrayOutputStream.html#close--) does not need to be closed:

> Closing a `ByteArrayOutputStream` has no effect. The methods in this class can be called after the stream has been closed without generating an `IOException`.

Reply:
Even a `close` call has no effect (regarding to the documentation) it would be a proper way to handle all stream objects in the same way and to use a `try-with-resources' statement should be possible to add.